tests/hypothesis_strategies/kernels.py

```diff
@@ -27,8 +51,66 @@ class KernelType(Enum):
 )
 """A strategy that generates Matern kernels."""
 
+periodic_kernels = st.builds(
+    PeriodicKernel,
+    lengthscale_prior=st.one_of(st.none(), priors),
+    lengthscale_initial_value=st.one_of(st.none(), finite_floats()),
+    period_length_prior=st.one_of(st.none(), priors),
+    period_length_initial_value=st.one_of(st.none(), finite_floats()),
+)
+"""A strategy that generates periodic kernels."""
+
+piecewise_polynomial_kernels = st.builds(
+    PiecewisePolynomialKernel,
+    q=st.integers(min_value=0, max_value=3),
+    lengthscale_prior=st.one_of(st.none(), priors),
+    lengthscale_initial_value=st.one_of(st.none(), finite_floats()),
+)
+"""A strategy that generates piecewise polynomial kernels."""
 
-base_kernels = st.one_of([matern_kernels])
+polynomial_kernels = st.builds(
+    PolynomialKernel,
+    power=st.integers(),
```

Comment:
I guess this needs to be 
```suggestion
    power=st.integers(min_value=0),
```
right?

Reply:
i dont think gpytorch limits it to positive numbers

Reply:
Oh, you mean that `x^-1` would also work? Interesting... but if that's the case, why then even the restriction to ints? And have you actually tested it? I think that's something we need to validate first.

Reply:
ive treid negatieve values not
-> does not throw valdiation or isntantiation error at runtine
-> seems to casue nuerical probems,mie `cholesky` or `could not fit any model` errors

what to do? restrict to >0 for now?

Reply:
In that case, I'd say we rather go with the restriction for now and only extend once needed (we then need to dive deeper in the corresponding PR) 👍🏼

Reply:
restricted as discussed